sweagent/frontend/src/Run.js

```diff
@@ -154,8 +157,44 @@ function Run() {
     }
   };
 
+  const checkScrollPosition = (ref, scrollStateRef, offset = 0) => {
+    scrollStateRef.current =
+      ref.current.scrollTop + ref.current.clientHeight + offset <
+      ref.current.scrollHeight;
+  };
+
+  const scrollToBottom = (ref, scrollStateRef) => {
+    if (!scrollStateRef.current) {
+      ref.current.scrollTop = ref.current.scrollHeight;
+    }
+  };
+
+  const scrollDetected = () => checkScrollPosition(logsRef, isLogScrolled, 58);
```

Comment:
why does the log need the offset and the others don't?

Reply:
probably also should be called `scrollLogDetected` to make the naming uniform

Reply:
The log needs the offset due to the copy to clipboard button that appears at the end. Updated the name to be scrollLogDetected